src/fhirtypes/common.ts

```diff
@@ -192,33 +506,68 @@ function traverseRulePathTree(elements: PathNode[]): string[] {
   });
   return result;
 }
-/**
- * Gets the id of an element using the shortcut syntax described here
- * https://blog.fire.ly/2019/09/13/type-slicing-in-fhir-r4/
- * @returns {string} the id for the shortcut
- */
-function getShortcutId(el: ElementDefinition): string {
-  return el.id
-    .split('.')
-    .map(p => {
-      const i = p.indexOf('[x]:');
-      const baseElementId = p.slice(0, i);
-      const choiceType = p.slice(i + baseElementId.length + 4);
-      const isChoiceSlice =
-        i > -1
-          ? CHOICE_TYPE_SLICENAME_POSTFIXES.includes(choiceType) &&
-            p === `${baseElementId}[x]:${baseElementId}${choiceType}`
-          : false;
-      return isChoiceSlice ? p.slice(i + 4) : p;
-    })
-    .join('.');
+
+type SliceNode = {
+  element: ElementDefinition;
+  children: SliceNode[];
+  count?: number;
+};
+
+function buildSliceTree(parent: ElementDefinition): SliceNode {
```

Comment:
@mint-thompson has done both in fact! Plus we moved the slice tree specific functions to their own file since they are so related to each other.

See [d14501d](https://github.com/FHIR/sushi/pull/1121/commits/d14501def6fe25249d86ebea2930c921bcc0ebc5).